src/worker/tasks/sendTransactionWorker.ts

```diff
@@ -400,18 +401,20 @@ const _resendTransaction = async (
     },
   });
 
-  // Double gas fee settings if they were not provded in `overrides`.
+  // Double the gas fee settings each attempt up to 10x.
+  // Do not update gas if overrides were provided.
+  const gasMultiple = BigInt(clamp(job.attemptsMade * 2, { min: 2, max: 10 }));
   if (populatedTransaction.gasPrice) {
-    populatedTransaction.gasPrice *= 2n;
+    populatedTransaction.gasPrice *= gasMultiple;
   }
   if (populatedTransaction.maxFeePerGas && !overrides?.maxFeePerGas) {
-    populatedTransaction.maxFeePerGas *= 2n;
+    populatedTransaction.maxFeePerGas *= gasMultiple;
   }
   if (
     populatedTransaction.maxPriorityFeePerGas &&
     !overrides?.maxPriorityFeePerGas
   ) {
-    populatedTransaction.maxPriorityFeePerGas *= 2n;
+    populatedTransaction.maxPriorityFeePerGas *= gasMultiple;
```

Comment:
nit:

`maxFeePerGas = maxBaseFeePerGas + maxPriorityFeePerGas`

Since `maxBaseFeePerGas` is uniform across all transactions in a block, setting it significantly above the current base fee provides no benefit. Any excess above the actual block's base fee cannot be collected.

Suggest focusing on increasing `maxPriorityFeePerGas` instead, as this:
1. Directly incentivizes validators to include the transaction
2. Can actually be collected (unlike excess base fee)

Consider reducing `maxBaseFeePerGas` multiplier and redistributing the margin to `maxPriorityFeePerGas` for more effective fee management.

```

// For EIP-1559 transactions
if (populatedTransaction.maxPriorityFeePerGas && !overrides?.maxPriorityFeePerGas) {
 const originalPriorityFee = populatedTransaction.maxPriorityFeePerGas;
 populatedTransaction.maxPriorityFeePerGas *= priorityFeeMultiple;
 
 // Adjust maxFeePerGas to accommodate the increased priority fee
 if (populatedTransaction.maxFeePerGas && !overrides?.maxFeePerGas) {
   const originalBaseFee = populatedTransaction.maxFeePerGas - originalPriorityFee;
   populatedTransaction.maxFeePerGas = originalBaseFee + populatedTransaction.maxPriorityFeePerGas;
 }
}
```